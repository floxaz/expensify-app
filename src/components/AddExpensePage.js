import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
   constructor(props) {
      super(props);
   }

   onSubmit = expense => {
      //props.dispatch(addExpense(expense));
      this.props.addExpense(expense); // remove it if you don't use mapDispatchToProps
      this.props.history.push('/');
   };

   render() {
      return (<div>
         <h2>Add Expense</h2>
         <ExpenseForm
            onSubmit={this.onSubmit}
         />
      </div>)
   }
}

// we wrote this cuz it's easier to to test
const mapDispatchToProps = dispatch => ({
   addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);