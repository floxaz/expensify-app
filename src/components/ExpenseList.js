import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = props => (
    <div>
       {props.expenses.length === 0 ? (
           <p>There are no expenses yet</p>
       ) : (
        props.expenses.map(expense => <ExpenseListItem key={expense.id} {...expense} />)
       )}
    </div>
);
// state is redux store we're passing
const mapStateToProps = state => {
    // here are the data this component can use
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

// HOC
export default connect(mapStateToProps)(ExpenseList);
