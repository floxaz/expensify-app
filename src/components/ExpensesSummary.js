import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = props => (
    <div>
      <p>Viewing {props.expenseCount} expenses totalling {props.expensesTotal}</p>
    </div>
);

const mapStateToProps = state => ({
    expenseCount: state.expenses.length,
    expensesTotal: selectExpensesTotal(state.expenses)
});

export default connect(mapStateToProps)(ExpenseSummary);