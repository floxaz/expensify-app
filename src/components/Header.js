import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
      <h1>Expensify</h1>
      <nav>
        <ul>
          <li><NavLink exact to="/" activeClassName="is-active">Dashboard</NavLink></li>
          <li><NavLink to="/create" activeClassName="is-active">Create expense</NavLink></li>
        </ul>
      </nav>
    </header>
);

export default Header;