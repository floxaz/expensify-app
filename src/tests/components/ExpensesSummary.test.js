import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenseSummary component with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary component with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={222453} />);
    expect(wrapper).toMatchSnapshot();
});