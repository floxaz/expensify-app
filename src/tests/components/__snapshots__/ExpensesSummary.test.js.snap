// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary component with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render ExpenseSummary component with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,224.53
  </h1>
</div>
`;
